File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/Polimorfismo/Ejer3(py y java)/Matriz.py
# a) Constructor matriz identidad
class Matriz:
    # a) y b) 
    def __init__(self, valores=None):
        self.matriz=[[0 for j in range(10)] for i in range(10)]
        if valores:
            # b)    
            for i in range(10):
                for j in range(10):
                    self.matriz[i][j]=valores[i][j]
        else:
            # a)
            for i in range(10):
                self.matriz[i][i]=1

    # c) 
    def sumar(self,otra):
        resultado=Matriz()
        for i in range(10):
            for j in range(10):
                resultado.matriz[i][j]=self.matriz[i][j]+otra.matriz[i][j]
        return resultado

    # c) 
    def restar(self,otra):
        resultado=Matriz()
        for i in range(10):
            for j in range(10):
                resultado.matriz[i][j]=self.matriz[i][j]-otra.matriz[i][j]
        return resultado

    # d) 
    def igual(self,otra):
        for i in range(10):
            for j in range(10):
                if self.matriz[i][j]!=otra.matriz[i][j]:
                    return False
        return True


    def mostrar(self):
        for fila in self.matriz:
            print(fila)

# d) 
m1=Matriz() 
m2=Matriz()
m3=m1.sumar(m2)
print("Suma:")
m3.mostrar()
print("Son iguales m1 y m2?", m1.igual(m2))

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/Polimorfismo/Ejer3(py y java)/Matriz.java
// a) y b) Clase Matriz
public class Matriz{
    private float[][] matriz;

    // a) 
    public Matriz(){
        matriz=new float[10][10];
        for(int i=0;i<10;i++){
            matriz[i][i]=1;
        }
    }

    // b)
    public Matriz(float[][] valores){
        matriz=new float[10][10];
        for(int i=0;i<10;i++){
            for(int j=0;j<10;j++){
                matriz[i][j]=valores[i][j];
            }
        }
    }

    // c)
    public Matriz sumar(Matriz otra){
        Matriz resultado=new Matriz();
        for(int i=0;i<10;i++){
            for(int j=0;j<10;j++){
                resultado.matriz[i][j]=this.matriz[i][j]+otra.matriz[i][j];
            }
        }
        return resultado;
    }

    // c) 
    public Matriz restar(Matriz otra){
        Matriz resultado=new Matriz();
        for(int i=0;i<10;i++){
            for(int j=0;j<10;j++){
                resultado.matriz[i][j]=this.matriz[i][j]-otra.matriz[i][j];
            }
        }
        return resultado;
    }

    // d) 
    public boolean igual(Matriz otra){
        for(int i=0;i<10;i++){
            for(int j=0;j<10;j++){
                if(this.matriz[i][j]!=otra.matriz[i][j]){
                    return false;
                }
            }
        }
        return true;
    }

    public void mostrar(){
        for(int i=0;i<10;i++){
            for(int j=0;j<10;j++){
                System.out.print(matriz[i][j]+" ");
            }
            System.out.println();
        }
    }

    // d) 
    public static void main(String[] args){
        Matriz m1=new Matriz();
        Matriz m2=new Matriz();
        Matriz m3=m1.sumar(m2);
        System.out.println("Suma:");
        m3.mostrar();
        System.out.println("Son iguales m1 y m2? "+m1.igual(m2));
    }
}
